src/kotlin/flwr/src/main/java/dev/flower/android/gRPC.kt

```diff
@@ -0,0 +1,236 @@
+        val request = PullTaskInsRequest.newBuilder().setNode(node).build()
+        requestChannel.send(request)
+    }
+
+    private suspend fun receive(requestChannel: Channel<PullTaskInsRequest>, node: Node) = flow {
+        coroutineScope {
+            val responses = Channel<TaskIns?>(1)
+            for (request in requestChannel)
+                asyncStub.pullTaskIns(request, object : StreamObserver<PullTaskInsResponse> {
+                    override fun onNext(value: PullTaskInsResponse?) {
+                        val taskIns = value?.let { getTaskIns(it) }
+                        if (taskIns != null && validateTaskIns(taskIns, true)) {
+                            state[KEYTASKINS] = taskIns
+                            responses.trySend(taskIns).isSuccess
+                        }
+
+                    }
+
+                    override fun onError(t: Throwable?) {
+                        t?.printStackTrace()
+                    }
+
+                    override fun onCompleted() {
+                    }
+                })
+
+            for (response in responses) {
+                if (response == null) {
+                    delay(3000)
+                    request(requestChannel, node)
+                } else {
+                    emit(response)
+                }
+            }
+        }
+    }
+
+    suspend fun startGrpcRere() {
+        createNode()
+
+        val node = nodeStore[KEYNODE]
+        if (node == null) {
+            println("Node not available")
+            return
+        }
+
+        val requestChannel = Channel<PullTaskInsRequest>(1)
+        request(requestChannel, node)
+        receive(requestChannel, node)
+            .collect {
+                val (taskRes, sleepDuration, keepGoing) = handle(client, it)
+                send(taskRes)
+                delay(sleepDuration.toLong())
+                if (keepGoing) {
+                    request(requestChannel, node)
+                } else {
+                    deleteNode()
+                    requestChannel.close()
+                }
+            }
+    }
+
+    private fun send(taskRes: TaskRes) {
+        nodeStore[KEYNODE]?.let { node ->
+            state[KEYTASKINS]?.let { taskIns ->
+                if (validateTaskRes(taskRes)) {
+                    taskRes
+                        .toBuilder()
+                        .setTaskId("")
+                        .setGroupId(taskIns.groupId)
+                        .setWorkloadId(taskIns.workloadId)
+                        .task.toBuilder()
+                        .setProducer(node)
+                        .setConsumer(taskIns.task.producer)
+                        .addAncestry(taskIns.taskId)
+                    val request = PushTaskResRequest.newBuilder().addTaskResList(taskRes).build()
+                    asyncStub.pushTaskRes(request, null)
+                }
+                state[KEYTASKINS] = null
+            }
+        }
+    }
+}
+
+suspend fun createFlowerReRe(
```

Comment:
Let's change `ReRe` everywhere to `Rere`:
```suggestion
suspend fun createFlowerRere(
```